cupy/cuda/compiler.py

```diff
@@ -318,18 +318,8 @@ def _jitify_prep(source, options, cu_path):
     return options, headers, include_names
 
 
-_has_usedforsecurity = (sys.version_info >= (3, 9))
-
-
-def _hash_hexdigest(value):
```

Comment:
Could you retain `_hash_hexdigest` and `_hash_length`? These functions exist so that we can easily change the algorithm in future when needed.

Reply:
I thought that was what you were saying here: https://github.com/cupy/cupy/pull/9053#issuecomment-2747426375?

Reply:
Sorry, my bad. I mistakenly thought you were referring to remove away `_has_usedforsecurity` flag.

To share the context, we once changed the hash algorithm from MD5 to SHA1 in the past (#5988), and later noticed that the hash length was not updated (which was fixed in #6023). To make the code robust to future changes, I'd prefer to keep using `_hash_hexdigest` and `_hash_length` here.

Reply:
Sure, I was just thinking you probably wouldn't change back since MD5 breaks FIPS :shrug:. I'll add it back.